src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/QPackEncoder.cs

```diff
@@ -477,9 +477,8 @@ private int EncodeStatusCode(int statusCode, Span<byte> buffer)
                 case 400:
                 case 404:
                 case 500:
-                    // TODO this isn't safe, some index can be larger than 64. Encoded here!
-                    buffer[0] = (byte)(0xC0 | H3StaticTable.StatusIndex[statusCode]);
-                    return 1;
+                    EncodeStaticIndexedHeaderField(H3StaticTable.StatusIndex[statusCode], buffer, out var bytesWritten);
+                    return bytesWritten;
```

Comment:
Can we keep single byte write for those statuses that are still a single byte? Should be faster for those.

Reply:
This is only called once per request. I'll leave it as-is because I don't want to micro-optimize too much. Will revisit header encoding/decoding later when focusing on perf.